examples/child_window.rs

```diff
@@ -1,37 +1,41 @@
-#[cfg(all(target_os = "linux", feature = "x11"))]
-use std::collections::HashMap;
+#[cfg(any(
+    all(target_os = "linux", feature = "x11"),
+    target_os = "macos",
+    target_os = "windows"
+))]
+fn main() {
+    use std::collections::HashMap;
 
-#[cfg(all(target_os = "linux", feature = "x11"))]
-use winit::{
-    dpi::{LogicalPosition, LogicalSize, Position},
-    event::{ElementState, Event, KeyboardInput, WindowEvent},
-    event_loop::{ControlFlow, EventLoop, EventLoopWindowTarget},
-    platform::x11::{WindowBuilderExtX11, WindowExtX11},
-    window::{Window, WindowBuilder, WindowId},
-};
+    use raw_window_handle::{HasRawWindowHandle, RawWindowHandle};
+    use winit::{
+        dpi::{LogicalPosition, LogicalSize, Position},
+        event::{ElementState, Event, KeyboardInput, WindowEvent},
+        event_loop::{ControlFlow, EventLoop, EventLoopWindowTarget},
+        window::{Window, WindowBuilder, WindowId},
+    };
 
-#[cfg(all(target_os = "linux", feature = "x11"))]
-fn spawn_child_window(
-    parent: u32,
-    event_loop: &EventLoopWindowTarget<()>,
-    windows: &mut HashMap<u32, Window>,
-) {
-    let child_window = WindowBuilder::new()
-        .with_parent(WindowId::from(parent as u64))
-        .with_title("child window")
-        .with_inner_size(LogicalSize::new(200.0f32, 200.0f32))
-        .with_position(Position::Logical(LogicalPosition::new(0.0, 0.0)))
-        .with_visible(true)
-        .build(event_loop)
-        .unwrap();
+    fn spawn_child_window(
+        parent: RawWindowHandle,
+        event_loop: &EventLoopWindowTarget<()>,
+        windows: &mut HashMap<WindowId, Window>,
+    ) {
+        // `with_parent_window` is unsafe. Parent window must a valid window.
+        let child_window = unsafe {
+            WindowBuilder::new()
+                .with_parent_window(Some(parent))
+                .with_title("child window")
+                .with_inner_size(LogicalSize::new(200.0f32, 200.0f32))
+                .with_position(Position::Logical(LogicalPosition::new(0.0, 0.0)))
+                .with_visible(true)
+                .build(event_loop)
+                .unwrap()
+        };
```

Comment:
Nit: Shrink `unsafe`

```suggestion
        let mut builder = WindowBuilder::new()
            .with_title("child window")
            .with_inner_size(LogicalSize::new(200.0f32, 200.0f32))
            .with_position(Position::Logical(LogicalPosition::new(0.0, 0.0)))
            .with_visible(true);
        builder = unsafe { builder.with_parent_window(Some(parent)) };
        let child_window = builder.build(event_loop).unwrap();
```